### PR TITLE
[bitnami/mariadb] New major version

### PR DESCRIPTION
**Pull request by juan131**
Signed-off-by: juan131 <juanariza@vmware.com>

**Description of the change**

This PR refactors the whole chart to follow Helm chart best practices. This includes several breaking changes, so please review it carefully, putting special emphasis to the "Upgrading" section describing all of them.

The most important change are:

- The terms *master* and *slave* have been replaced by the terms *primary* and *secondary*.
- The default MariaDB version was updated from `10.3` to `10.5`.
- Labels are adapted to follow the [Helm charts standard labels](https://helm.sh/docs/chart_best_practices/labels/#standard-labels).

**Benefits**

- Standardization.
- Reduces technical debt.

**Benefits**

**Possible drawbacks**

It breaks backwards compatibility.

**Applicable issues**

None

**Additional information**

Upgrade tests from version `7.x.x` using workaround suggested in the Upgrading Notes:

- Install old chart version:

```console
$ helm install mariadb bitnami/mariadb \
    --set rootUser.password=root_password \
    --set replication.enabled=true \
    --set replication.password=rep_password \
    --set db.user=my_username \
    --set db.name=my_database \
    --set db.password=my_password \
    --version 7.10.3
```

- Add some data once everything is up and running:

```console
$ kubectl run mariadb-client --rm --tty -i --restart='Never' --image  docker.io/bitnami/mariadb:10.3.24-debian-10-r17 --namespace default --command -- bash
I have no name!@mariadb-client:/$ mysql -h mariadb.default.svc.cluster.local -uroot -p my_database
Enter password:
...
MariaDB [my_database]> CREATE OR REPLACE TABLE foo (a int);
Query OK, 0 rows affected (0.032 sec)
MariaDB [my_database]> INSERT INTO foo (a) VALUES (10);
Query OK, 1 row affected (0.018 sec)
```

- Remove chart and slave PVC (we only need the master PVC since slave only replicates data):

```console
$ helm uninstall mariadb && kubectl delete pvc data-mariadb-slave-0
$ kubectl get pvc
NAME                    STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
data-mariadb-master-0   Bound    pvc-d9b6a420-146a-4fad-94eb-9dbace8cacd4   8Gi        RWO            standard       6m49s
```

- Install new chart version reusing existing PVC for primary node:

```console
$ helm install mariadb bitnami/mariadb \
  --set architecture=replication \
  --set auth.rootPassword=root_password \
  --set auth.replicationPassword=rep_password \ 
  --set auth.username=my_username \
  --set auth.password=my_password \
  --set auth.database=my_database \
  --set primary.persistence.existingClaim=data-mariadb-master-0 \
  --version 8.0.0
```

- Ensure the data is reused/upgraded (MariaDB 10.3 to 10.5):

```console
$ kubectl logs mariadb-primary-0
...
mariadb 11:15:55.89 INFO  ==> Using persisted data
mariadb 11:15:55.91 INFO  ==> Running mysql_upgrade
...
```

- Check data is okay:

```console

$ kubectl run mariadb-client --rm --tty -i --restart='Never' --image  docker.io/bitnami/mariadb:10.5.5-debian-10-r48 --namespace default --command -- bash
I have no name!@mariadb-client:/$ mysql -h mariadb-secondary.default.svc.cluster.local -uroot -p my_database
Enter password:
...
MariaDB [my_database]> SELECT * FROM foo;
+------+
| a    |
+------+
|   10 |
+------+
1 row in set (0.002 sec)
MariaDB [my_database]> SHOW GRANTS FOR 'my_username';
+------------------------------------------------------------------------------------------------------------+
| Grants for my_username@%                                                                                   |
+------------------------------------------------------------------------------------------------------------+
| GRANT USAGE ON *.* TO `my_username`@`%` IDENTIFIED BY PASSWORD '*CCD3A959D6A004B9C3807B728BC2E55B67E10518' |
| GRANT ALL PRIVILEGES ON `my_database`.* TO `my_username`@`%`                                               |
+------------------------------------------------------------------------------------------------------------+
2 rows in set (0.001 sec)
```

**Checklist**

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

File: bitnami/mariadb/templates/_helpers.tpl [hunk 1]
```diff
@@ -1,108 +1,53 @@
...
-Also, we can't use a single if because lazy evaluation is not an option
+Return the appropriate apiVersion for statefulset.
 */}}
-{{- if .Values.global }}
-    {{- if .Values.global.imageRegistry }}
-        {{- printf "%s/%s:%s" .Values.global.imageRegistry $repositoryName $tag -}}
-    {{- else -}}
-        {{- printf "%s/%s:%s" $registryName $repositoryName $tag -}}
-    {{- end -}}
+{{- define "mariadb.statefulset.apiVersion" -}}
```
Comment (dani8art) [inline bitnami/mariadb/templates/_helpers.tpl:46]: could it be moved to the common too?

**Review by dani8art — CHANGES_REQUESTED** [review]
Some comments about password validations

File: bitnami/mariadb/templates/NOTES.txt [hunk 2]
```diff
@@ -41,9 +41,22 @@ To upgrade this helm chart:
...
+    {{- $requiredDBPassword := dict "valueKey" "auth.password" "secret" $secretName "field" "mariadb-password" "context" $ -}}
+    {{- $requiredPasswords = append $requiredPasswords $requiredDBPassword -}}
+  {{- end -}}
+  {{- if eq .Values.architecture "replication" }}
+      {{- $requiredReplicationPassword := dict "valueKey" "auth.replicationPassword" "secret" $secretName "field" "mariadb-replication-password" "context" $ -}}
+      {{- $requiredPasswords = append $requiredPasswords $requiredReplicationPassword -}}
+  {{- end -}}
+  {{- $requiredPasswordValidationErrors := include "common.validations.values.multiple.empty" (dict "required" $requiredPasswords "context" $) -}}
+  {{- include "common.errors.upgrade.passwords.empty" (dict "validationErrors" (list $requiredPasswordValidationErrors) "context" $) -}}
+{{- end -}}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/mariadb/templates/NOTES.txt:62]: See https://github.com/bitnami/charts/pull/3859

File: bitnami/mariadb/templates/_helpers.tpl (hunk 1 above)
Comment (juan131) [reply bitnami/mariadb/templates/_helpers.tpl:46]: See https://github.com/bitnami/charts/pull/3859

**Review by dani8art — APPROVED** [review]
LGTM

**Pull request merged**